### PR TITLE
Sort objects inside arrays when using `deep` option

### LINKED ISSUE #5: Objects inside arrays still need sorting
I had to write [my own sorter](https://github.com/kartotherian/babel/blob/master/test/debug-utils.js#L31), but would be great if you added this to your code - an ability to sort objects inside arrays, without changing the position in the arrays themselves.
```js
{
  b: 1
  a: [ { f: 10, e: 11 }, { h: 20, g: 21 }]
}
```
after sort:
```js
{
  a: [ { e: 11, f: 10 }, { g: 21, h: 20 }]
  b: 1
}
```

### PR DESCRIPTION
**Pull request by kevva**
Fixes #5.

### CODE REVIEW

File: test.js [hunk 1]
```diff
@@ -47,4 +47,5 @@ test('deep option', t => {
 	t.is(sorted, sorted.a.c);
 	t.deepEqual(Object.keys(sorted), ['a', 'b']);
 	t.deepEqual(m({c: {c: 0, a: 0, b: 0}, a: 0, b: 0, z: [9, 8, 7, 6, 5]}, {deep: true}), {a: 0, b: 0, c: {a: 0, b: 0, c: 0}, z: [9, 8, 7, 6, 5]});
+	t.deepEqual(Object.keys(m({a: [{b: 0, a: 0}]}, {deep: true}).a[0]), ['a', 'b']);
```
Comment (sindresorhus) [inline test.js]: Can you add a test to ensure it handles circular references in deep arrays too?

Comment (kevva) [inline test.js]: Forgot about it. Should be fixed now!

File: index.js [hunk 2]
```diff
@@ -34,6 +34,17 @@ module.exports = function (obj, opts) {
 			var key = keys[i];
 			var val = x[key];
 
+			if (deep && Array.isArray(val)) {
+				var retArr = [];
+
+				for (var j = 0; j < val.length; j++) {
```
Comment (dylansmith) [inline index.js]: Might be preferable to use `ret[key] = val.map(i => isPlainObj(i) ? sortKeys(i) : i` ?

Comment (kevva) [inline index.js]: Yeah, I tried keeping it as close as possible to the current code style wise. [This](https://github.com/sindresorhus/sort-keys/blob/f977719f258f4e117be7821e3ed70e7c99468f6e/index.js#L33-L52) whole part can be rewritten.

Comment (sindresorhus) [inline index.js]: Happy to accept a follow-up refactor PR.

**Pull request merged**